openapi-cli/src/main/java/io/ballerina/openapi/generators/client/BallerinaClientGenerator.java

```diff
@@ -200,6 +200,11 @@ public SyntaxTree generateSyntaxTree() throws BallerinaOpenApiException {
             if (ballerinaAuthConfigGenerator.isApiKey() && ballerinaAuthConfigGenerator.isHttpOROAuth()) {
                 nodes.add(ballerinaAuthConfigGenerator.getAuthConfigRecord());
             }
+
+            // Add custom `OAuth2ClientCredentialsGrantConfig` record with default tokenUrl if `tokenUrl` is available
+            if (ballerinaAuthConfigGenerator.getClientCredGrantTokenUrl() != null) {
+                nodes.add(ballerinaAuthConfigGenerator.getOAuth2ClientCredsGrantConfigRecord());
+            }
```

Comment:
Shall we write a single function at `BallerinaAuthConfigGenerator` to get all 3 records at once since all these if conditions check flags available at `BallerinaAuthConfigGenerator`.

Reply:
+1. Done